Add unit test folder (#698)
pypa/packaging.python.org: source/tutorials/packaging-projects.rst

    packaging_tutorial/
      example_pkg/
        __init__.py

>       test/

Reviewer: ```suggestion
      tests/
```

      setup.py
      LICENSE
      README.md

Creating a test folder
----------------------

> :file:`test/` is a placeholder for unit test files. Leave it empty for now.

Reviewer: ```suggestion

:file:`tests/` is a placeholder for unit test files. Leave it empty for now.
```

    packaging_tutorial/
      example_pkg/
        __init__.py
      tests/
      setup.py
      LICENSE
      README.md

Creating a test folder
----------------------

> :file:`tests/` is a placeholder for unit test files. Leave it empty for now.

Reviewer 2: ```suggestion
:file:`tests/` is a placeholder for unit test files. Leave it empty for now.

  .. note::

     In *rare* cases, you may meet some projects that put the ``tests/`` folder under
     ``example_pkg/``. They probably use `pytest's pyargs`_. While there may be a
     legitimate reason to do so, you'll likely never need it. So for the sake of
     simplicity, this guide will stick with the most common case of a top-level
     ``tests/`` dir (in plural).

     .. _pytest's pyargs:
     https://docs.pytest.org/en/latest/goodpractices.html#tests-as-part-of-application-code
```

@pfmoore ^

Reviewer 3: That seems too wordy to me, in that it obscures the key point. Maybe just 

```
:file:`tests/` is a placeholder for unit test files (other test layouts are possible, but this is a good starting point). Leave it empty for now.
```

?

Reviewer 2: Yeah, that sounds good. Should we keep the link?

```suggestion
:file:`tests/` is a placeholder for test files (`other test layouts`_ are possible, but this is a good starting point). Leave it empty for now.

.. _other test layouts:
   https://docs.pytest.org/en/latest/goodpractices.html#tests-as-part-of-application-code
```

@pfmoore ^

Reviewer 3: I'd argue no. If we link to a discussion of that layout, we should also link to a discussion of the top-level `tests` layout, and once again we spiral into bikeshedding.

> :file:`tests/` is a placeholder for unit test files. Leave it empty for now.

Reviewer 2: Nitpick: I'd probably remove "unit" from there. Because other types of tests exist and can be put there.
